src/proxyvisor/proxyvisor.coffee

```diff
@@ -1,232 +1,35 @@
-					return 'status must be a non-empty string'
-				if !validStringOrUndefined(commit)
-					return 'commit must be a non-empty string'
-				if !validStringOrUndefined(releaseId)
-					return 'commit must be a non-empty string'
-				if !validObjectOrUndefined(environment)
-					return 'environment must be an object'
-				if !validObjectOrUndefined(config)
-					return 'config must be an object'
-				return null
-			requestError = validateDeviceFields()
-			if requestError?
-				res.status(400).send(requestError)
-				return
-
-			environment = JSON.stringify(environment) if isDefined(environment)
-			config = JSON.stringify(config) if isDefined(config)
-
-			fieldsToUpdateOnDB = _.pickBy({ status, is_online, commit, releaseId, config, environment }, isDefined)
-			fieldsToUpdateOnAPI = _.pick(fieldsToUpdateOnDB, 'status', 'is_online', 'commit', 'releaseId')
-
-			if _.isEmpty(fieldsToUpdateOnDB)
-				res.status(400).send('At least one device attribute must be updated')
-				return
-
-			@db.models('dependentDevice').select().where({ uuid })
-			.then ([ device ]) =>
-				return res.status(404).send('Device not found') if !device?
-				return res.status(410).send('Device deleted') if device.markedForDeletion
-				throw new Error('Device is invalid') if !device.deviceId?
-				Promise.try =>
-					if !_.isEmpty(fieldsToUpdateOnAPI)
-						@proxyvisor.apiBinder.patchDevice(device.deviceId, fieldsToUpdateOnAPI)
-				.then =>
-					@db.models('dependentDevice').update(fieldsToUpdateOnDB).where({ uuid })
-				.then ->
-					res.json(parseDeviceFields(device))
-			.catch (err) ->
-				console.error("Error on #{req.method} #{url.parse(req.url).pathname}", err, err.stack)
-				res.status(503).send(err?.message or err or 'Unknown error')
-
-		@router.get '/v1/dependent-apps/:appId/assets/:commit', (req, res) =>
-			@db.models('dependentApp').select().where(_.pick(req.params, 'appId', 'commit'))
-			.then ([ app ]) =>
-				return res.status(404).send('Not found') if !app
-				dest = tarPath(app.appId, app.commit)
-				fs.lstatAsync(dest)
-				.catch =>
-					Promise.using @docker.imageRootDirMounted(app.image), (rootDir) ->
-						getTarArchive(rootDir + '/assets', dest)
-				.then ->
-					res.sendFile(dest)
-			.catch (err) ->
-				console.error("Error on #{req.method} #{url.parse(req.url).pathname}", err, err.stack)
-				res.status(503).send(err?.message or err or 'Unknown error')
-
-		@router.get '/v1/dependent-apps', (req, res) =>
-			@db.models('dependentApp').select()
-			.map (app) ->
-				return {
-					id: parseInt(app.appId)
-					commit: app.commit
-					name: app.name
-					config: JSON.parse(app.config ? '{}')
-				}
-			.then (apps) ->
-				res.json(apps)
-			.catch (err) ->
-				console.error("Error on #{req.method} #{url.parse(req.url).pathname}", err, err.stack)
-				res.status(503).send(err?.message or err or 'Unknown error')
+		@router.use('/v1', new proxyvisorRouterV1.ProxyvisorRouterV1(@proxyvisor))
```

Comment:
Continuing from the comment above, here you can just use `new ProxyvisorRouterV1(@proxyvisor)` instead of `new proxyvisorRouterV1.ProxyvisorRouterV1(@proxyvisor)`